### ISSUE TITLE
non-admin user has no permissions on system32\Tasks

### ISSUE THREAD

**Issue by Adriaantje** [SandboxEscaper/polarbearrepo#2]
I fiddled around with your PoC a bit, but it seems to fail because a non-admin user has no delete permissions on the system32\Tasks folder (at least on the systems I tried this on)

After running schtasks for the first time, it tries to delete the task that was just created, but I get access denied. After that, the whole thing fails.

Permissions on the Tasks folder look like the following for me:
```
Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

C:\Windows\System32>icacls Tasks
Tasks BUILTIN\Administrators:(CI)(F)
      BUILTIN\Administrators:(OI)(R,W,D,WDAC,WO)
      NT AUTHORITY\SYSTEM:(CI)(F)
      NT AUTHORITY\SYSTEM:(OI)(R,W,D,WDAC,WO)
      NT AUTHORITY\Authenticated Users:(CI)(W,Rc)
      NT AUTHORITY\NETWORK SERVICE:(CI)(W,Rc)
      NT AUTHORITY\LOCAL SERVICE:(CI)(W,Rc)
      CREATOR OWNER:(OI)(CI)(IO)(F)

Successfully processed 1 files; Failed processing 0 files
```

Is there anyway this can still work like this, or is this actually the remedy?

**Comment by MrPurple0** [comment]
> I fiddled around with your PoC a bit, but it seems to fail because a non-admin user has no delete permissions on the system32\Tasks folder (at least on the systems I tried this on)
> 
> After running schtasks for the first time, it tries to delete the task that was just created, but I get access denied. After that, the whole thing fails.
> 
> Permissions on the Tasks folder look like the following for me:
> 
> ```
> Microsoft Windows [Version 10.0.14393]
> (c) 2016 Microsoft Corporation. All rights reserved.
> 
> C:\Windows\System32>icacls Tasks
> Tasks BUILTIN\Administrators:(CI)(F)
>       BUILTIN\Administrators:(OI)(R,W,D,WDAC,WO)
>       NT AUTHORITY\SYSTEM:(CI)(F)
>       NT AUTHORITY\SYSTEM:(OI)(R,W,D,WDAC,WO)
>       NT AUTHORITY\Authenticated Users:(CI)(W,Rc)
>       NT AUTHORITY\NETWORK SERVICE:(CI)(W,Rc)
>       NT AUTHORITY\LOCAL SERVICE:(CI)(W,Rc)
>       CREATOR OWNER:(OI)(CI)(IO)(F)
> 
> Successfully processed 1 files; Failed processing 0 files
> ```
> 
> Is there anyway this can still work like this, or is this actually the remedy?

Same issue occurred with me and im actually testing this on a spare live machine. Do you think it has something to do with the code being for 32bit machines? Because it was stated in the readme file 
"How to run the PoC
(**you need to rebuild for x64**, _included binary is x86_)".
Please shed some light on this.

**Comment by Adriaantje** [comment]
Oh, I forgot to mention. I did rebuild it for x64.

I also tried running the commands manually from cmd, gave me the same result.

**Comment by MrPurple0** [comment]
or maybe im still doing something wrong. even tried with program compatibility lol, same thing

**Comment by Sn00z3r** [comment]
Non-administrators do not have the necessary rights on both windows/tasks and windows/system32/tasks, so you need to be in the local Administrators group for it to work. So as far as I can see, this is not a priv esc bug, as you can get system through a myriad of options if you are an administrator

**Comment by MrPurple0** [comment]
But isn't the point of the exploit? Priv escalation for non admin users? And also looking back at the result demo, it did  show that sandboxesc was using a non-admin user account.

**Comment by Sn00z3r** [comment]
Where did it show that? I can't see any indication that the account is not in the local administrators group. There was no "net user test2" shown.

**Comment by HenkPoley** [comment]
Elsewhere people say that Windows 10 17xx was vulnerable, but newer versions are not.

**Comment by misuchiru03** [comment]
The proof was in the video when user `test` right clicked the driver and it showed a "run as admin" icon next to "Delete", then once the exploit ran, the right click menu did not show the icon anymore.

I do agree that the POC video needed a little more proof/context.

**Comment by Sn00z3r** [comment]
That icon you get when you are in the admin group as well. It's UAC, so it proves nothing :).

**Comment by misuchiru03** [comment]
> That icon you get when you are in the admin group as well. It's UAC, so it proves nothing :).
> 
> ![image](https://user-images.githubusercontent.com/20971050/58315632-2eba8e80-7e12-11e9-859a-d6c769f67ddf.png)

Hence the quotes around "proof".  However, if the UAC icon is gone, that would mean that the user now has ownership and full permissions to the item in question... basically admin, not just part of the administrators group.  I completely agree, however, there is little proof that the user wasn't already capable of doing this without the exploit as an admin can take ownership of stuff.

![UACicons](https://user-images.githubusercontent.com/22798851/58375275-82150580-7f14-11e9-91bf-03e39a2bc95d.png)
![OriginalPermissions](https://user-images.githubusercontent.com/22798851/58375271-6dd10880-7f14-11e9-8cc2-ea5b8b74cfa7.png)
![OriginalOwnership](https://user-images.githubusercontent.com/22798851/58375272-70336280-7f14-11e9-870c-92fb12c8feb5.png)
![NewOwnership](https://user-images.githubusercontent.com/22798851/58375273-73c6e980-7f14-11e9-8bd2-9d92fecbbb83.png)
![NewPermissions](https://user-images.githubusercontent.com/22798851/58375274-7590ad00-7f14-11e9-8e53-3d1ec3445e34.png)
![NoUACIcons](https://user-images.githubusercontent.com/22798851/58375269-614cb000-7f14-11e9-850a-f70c2ed3e171.png)
